battery_management_system/measurement_units.py

```diff
@@ -0,0 +1,30 @@
+from abc import ABC, abstractclassmethod
+from unit_conversion import get_celcius_from_fahrenheit, get_fahrenheit_from_celcius
+
+
+class Units(ABC):
+    @abstractclassmethod
+    def convert_to_celcius(self, unit):
+        raise NotImplementedError
+
+    def get_accepted_units():
+        return [unit.__name__ for unit in Units.__subclasses__()]
+
+
+class Celcius(Units):
+    def __init__(self, celcius) -> None:
+        self.celcius = celcius
+
+    def convert_to_celcius(self):
```

Comment:
class Celcius and Fahrenheit are not similar, in the sense, convert to self is only present in Celcius.

Reply:
>  You have considered units for temp as celcius, unit for SOC as Fahrenheit, and charge rate as celcius.

I have done this to showcase that the app can work on any measurement, **that it will be more flexible** 
the MIN and MAX component can be set in Fahrenheit or Celcius and similarly the input value can be either Fahrenheit or Celcius.
The app works in all case

Reply:
@prithamvshetty
I have modified as expected, could you please review now!
Thanks in advance!